signers/legacy/search.go

```diff
@@ -0,0 +1,238 @@
+package legacy
+
+import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"github.com/acquia/http-hmac-go/signers"
+	"hash"
+	"io/ioutil"
+	"log"
+	"os"
+	"net/http"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+var logger = log.New(os.Stdout, "", log.LstdFlags | log.Lshortfile)
+
+type SearchSigner struct {
+	*signers.Digester
+	*signers.Identifiable
+	respSigner *SearchResponseSigner
+}
+
+func NewSearchSigner(digest func() hash.Hash) (*SearchSigner, *signers.AuthenticationError) {
+	re, err := regexp.Compile("(?i)^\\s*acquia_solr_time.*?$")
+	if err != nil {
+		return nil, signers.Errorf(500, signers.ErrorTypeInternalError, "Could not compile regular expression for identifier: %s", err.Error())
+	}
+
+	return &SearchSigner{
+		Digester: &signers.Digester{
+			Digest: digest,
+		},
+		Identifiable: &signers.Identifiable{
+			IdRegex: re,
+		},
+		respSigner: NewSearchResponseSigner(digest),
+	}, nil
+}
+
+// This is not used, required by the interface
+func (v *SearchSigner) HashBody(r *http.Request) (string, *signers.AuthenticationError) {
+	data, err := signers.ReadBody(r)
+	if err != nil {
+		return "", signers.Errorf(500, signers.ErrorTypeInternalError, "Failed to read request body: %s", err.Error())
+	}
+	return v.HashBytes(data), nil
+}
+
+// This is not used, required by the interface
+func (v *SearchSigner) HashBytes(b []byte) string {
+	//h := sha256.New()
+	//h.Write(b)
+	//return base64.StdEncoding.EncodeToString(h.Sum(nil))
+	return ""
+}
+
+func (v *SearchSigner) GetResponseSigner() signers.ResponseSigner {
+	return v.respSigner
+}
+
+func (v *SearchSigner) Sign(r *http.Request, authHeaders map[string]string, secret string) (string, *signers.AuthenticationError) {
+
+	var hash string
+	var path_and_query string
+	var request_time int64
+
+	// get / validate headers
+	auth_headers := v.ParseAuthHeaders(r)
+    if auth_headers["acquia_solr_time"] == "" {
```

Comment:
This validation might fit better in the ParseAuthHeaders function so that you don't have to do it again down below.  Only works if ParseAuthHeaders is not part of the interface and you can change it to do 'err, auth_headers := v.ParseAuthHeaders(r)'